File: scripts/update_instock.js
const fs = require('fs');
const path = require('path');

const PRODUCTS = path.join(__dirname, '..', 'products.json');
const choices = ['high', 'medium', 'low', 'out'];

function main() {
  const raw = fs.readFileSync(PRODUCTS, 'utf8');
  let data;
  try {
    data = JSON.parse(raw);
  } catch (err) {
    console.error('Failed to parse products.json:', err.message);
    process.exit(1);
  }

  for (const item of data) {
    item.inStock = choices[Math.floor(Math.random() * choices.length)];
  }

  fs.writeFileSync(PRODUCTS, JSON.stringify(data, null, 2), 'utf8');
  console.log(`Updated ${data.length} products in ${PRODUCTS}`);
}

if (require.main === module) main();

File: scripts/update_instock.py
import json
import random
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
PRODUCTS = ROOT / 'products.json'

def main():
    choices = ["high", "medium", "low", "out"]
    with PRODUCTS.open('r', encoding='utf-8') as f:
        data = json.load(f)

    for item in data:
        item['inStock'] = random.choice(choices)

    with PRODUCTS.open('w', encoding='utf-8') as f:
        json.dump(data, f, indent=2, ensure_ascii=False)

if __name__ == '__main__':
    main()
